packages/diagrams/backend/sirius-components-collaborative-diagrams/src/main/java/org/eclipse/sirius/components/collaborative/diagrams/handlers/InvokeSingleClickOnDiagramElementToolEventHandler.java

```diff
@@ -172,4 +165,32 @@ private IStatus executeTool(IEditingContext editingContext, IDiagramContext diag
         return result;
     }
 
+    private Optional<Object> getCurrentContext(IEditingContext editingContext, String diagramElementId, SingleClickOnDiagramElementTool tool, Diagram diagram, Optional<Node> node,
+            Optional<Edge> edge) {
+        Optional<Object> self = Optional.empty();
+        if (node.isPresent()) {
+            self = this.objectService.getObject(editingContext, node.get().getTargetObjectId());
+        } else if (edge.isPresent()) {
+            self = this.objectService.getObject(editingContext, edge.get().getTargetObjectId());
+        } else if (Objects.equals(diagram.getId(), diagramElementId)) {
+            self = this.objectService.getObject(editingContext, diagram.getTargetObjectId());
+        } else {
+            this.logger.warn("The node creation tool {0} cannot be applied on the current diagram {1} and editing context {2}", tool.getId(), diagram.getId(), editingContext.getId());
```

Comment:
"The node creation tool" part of the message is even less relevant now. Just "The tool" is probably enough and less confusing.

Reply:
Done